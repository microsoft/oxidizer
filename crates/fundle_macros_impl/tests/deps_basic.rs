// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use syn::{ItemStruct, parse_quote};

mod util;

#[test]
fn basic_expansion() {
    let item: ItemStruct = parse_quote! {
        #[deps]
        struct Foo {
            x: String
        }
    };

    insta::assert_snapshot!(expand_fundle_deps!(item));
}
