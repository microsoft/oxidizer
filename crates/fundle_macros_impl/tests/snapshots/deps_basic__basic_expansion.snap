---
source: crates/fundle_macros_impl/tests/deps_basic.rs
assertion_line: 16
expression: expand_fundle_deps!(item)
---
#[allow(private_bounds)]
struct Foo {
    x: String,
}
#[allow(private_bounds)]
impl<T> From<T> for Foo
where
    T: AsRef<String>,
{
    fn from(value: T) -> Self {
        Self {
            x: <T as AsRef<String>>::as_ref(&value).to_owned(),
        }
    }
}
